lib/engine.js

```diff
@@ -17,80 +17,95 @@
 \*───────────────────────────────────────────────────────────────────────────*/
 'use strict';
 
-var path = require('path'),
-    dust = require('dustjs-linkedin'),
-    utils = require('./utils'),
-    patch = require('./patch'),
-    River = require('./brook'),
-    reqwire = require('./reqwire');
+var freshy = require('freshy').freshy;
+var path = require('path');
+var utils = require('./utils');
+var River = require('./brook');
+var reqwire = require('./reqwire');
+var dustjacket = require('dustjacket');
+var fs = require('fs');
 
 
-var loaders = {
-    dust: require('./reader/dust'),
-    js: require('./reader/js')
-};
+function fileLoader(name, context, cb) {
+    var views = utils.resolveViewDir(context.global);
 
+    var ext = path.extname(name) || 'dust';
+    if (ext[0] !== '.') {
+        ext = '.' + ext;
+    }
 
+    var file = path.resolve(views, name + ext);
+    fs.readFile(file, 'utf8', function (err, content) {
+        if (err) {
+            if (err.code !== 'ENOENT') {
+                cb(err);
+            } else {
+                cb();
```

Comment:
Does this have the potential for subtle or hard to debug issues with missing or misnamed templates? If I'm reading this correctly, missing files are effectively a noop (moving on to the next "middleware" registered with `dustjacket`) but here `fileLoader` is the base-case, so what happens?